### DISCUSSION TITLE
How to use with TypeScript?

### DISCUSSION THREAD

**Discussion by ngocdaothanh** [vip-git/react-jsonschema-form-material-ui#363] [Ideas]
Hi, please provide TypeScript types?

**Comment by vip-git** [comment]
Hi @ngocdaothanh 

Currently support for typescript is in progress as part of finalizing [version 3](https://github.com/vip-git/react-jsonschema-form-material-ui/releases). Expect that to be released in coming weeks.

Would you like to help in contributing some types as well ?

Looking forward to your reply.

Thanks

**Reply by ngocdaothanh** [reply]
Is it easier to just convert the project to use TypeScript, instead of writing the types manually?

**Reply by ngocdaothanh** [reply]
In case of switching to TypeScript, should we use Google gts instead of using webpack manually?
https://github.com/google/gts

**Reply by vip-git** [reply]
Yes the plan is to convert the project to use typescript i am planning to add that support part of `3.0.1` release. 

Thanks for the link on `gts` it seems have a open vulnerability currently
- https://snyk.io/test/github/google/gts 

`Gts` under water uses `eslint` and with eslint this parser / linter seems to be the most mature of all branched from `tslint`
- https://github.com/typescript-eslint/typescript-eslint

I will be adding support for this. But would love to know your thoughts why [gts](https://github.com/google/gts) would be better preferred over [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint)

**Reply by ngocdaothanh** [reply]
> why gts would be better preferred over typescript-eslint

gts is not only a linter.
Conceptually, it's like a `create-react-app`, it can also build the TypeScript project.
It's simple to create a new TypeScript project with it, without having to add configuration files.

So if `react-jsonschema-form-material-ui` doesn't require complex configuration, maybe gts is suitable.

**Reply by vip-git** [reply]
Yes although we are dependant on `gts` to keep updating following [dependencies](https://github.com/google/gts/blob/main/package.json#L46-L47) for us. 

Although in conclusion we can give this a try - i like the way its structured although [not updated frequently](https://github.com/google/gts/pulls) incase if we start facing issues we can revert back to vanilla eslint config 👍 

I will keep you posted on the PR in coming days for typescript inclusion.

If you would have more additions to suggest please do so :) 

Thanks
